Source/Core/Common/Config/Config.cpp

```diff
@@ -5,51 +5,61 @@
 #include <algorithm>
 #include <list>
 #include <map>
+#include <shared_mutex>
 
 #include "Common/Config/Config.h"
 
 namespace Config
 {
+using Layers = std::map<LayerType, std::shared_ptr<Layer>>;
+
+static std::shared_mutex s_layersRWLock;
```

Comment:
```suggestion
static std::shared_mutex s_layers_rw_lock;
```

We use `snake_case` as our naming scheme in new code